website/ember_osf_web/decorators.py

```diff
@@ -12,6 +12,17 @@ class MockUser(object):
     is_authenticated = False
 
 
+def camouflage_request(request):
+    """
+    This function changes the typical flask request object so it can be used by django-waffle. Other modifications for
+    django-waffle can be found in OsfWebRenderer.
+    :param flask request object:
+    :return flask request object:
+    """
+    request.COOKIES = request.cookies
```

Comment:
This should set `request.user`, too.
Also, delete the lines where `request.user` was previously set elsewhere. (I found a few: [1](https://github.com/CenterForOpenScience/osf.io/pull/9084/files#diff-f29cfbc5c72a696cd71d4c3ff0c35e73L24), [2](https://github.com/CenterForOpenScience/osf.io/pull/9084/files#diff-f29cfbc5c72a696cd71d4c3ff0c35e73L35), [3](https://github.com/CenterForOpenScience/osf.io/pull/9084/files#diff-f29cfbc5c72a696cd71d4c3ff0c35e73L40))
```suggestion
    request.COOKIES = request.cookies
    request.user = _get_current_user() or MockUser()
```